src/Tickets/Seating/Service/Layouts.php

```diff
@@ -125,7 +128,41 @@ public function get_in_option_format() {
 
 		return $layouts;
 	}
-
+	
+	/**
+	 * Fetches all the Layouts from the database.
+	 *
+	 * @since TBD
+	 *
+	 * @return Layout_Card[] Array of layout card objects.
+	 */
+	public function get_in_card_format() {
+		if ( ! $this->update() ) {
+			return [];
+		}
+		
+		$mem_key      = 'option_layout_card_objects';
+		$cache        = tribe_cache();
+		$layout_cards = $cache[ $mem_key ];
+		
+		if ( ! ( $layout_cards && is_array( $layout_cards ) ) ) {
+			$layout_cards = [];
+			foreach ( Layouts_Table::fetch_all() as $row ) {
+				$layout_cards[] = new Layout_Card(
+					$row->id,
+					$row->name,
+					$row->map,
+					$row->seats,
+					$row->screenshot_url
+				);
+			}
```

Comment:
If we have it as a factory style method we could do something like:
```
$cache[ $mem_key ] = array_map( [ Layout_Card::class, 'from_row' ], Layouts_Table::fetch_all() );
```

Reply:
The fetch method from table is a generator so we can't feed that directly to [`array_map`](https://share.cleanshot.com/69kW7XhR).

For now, we should settle with this structure and later I will try to use the generator method in future PRs and try to simplify this one as possible.

Reply:
We're using a Generator to limit the number of stuff pulled from the db and the number of items to keep in memory.
The factory method would require, under the hood, an unroll of the Generator with an `iterator_to_array` call that would still pull items in batches (good), but would store all of them in memory (bad).